### PR TITLE
Server/Component convertors

### PR DESCRIPTION
**Pull request by joelrebel**
Implements methods and defines a few consts to deal with unserializing server attributes 
relevant https://github.com/metal-toolbox/mctl/pull/79

### CODE REVIEW

File: serverservice/methods.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		}
+
+		for _, vattr := range c.VersionedAttributes {
+			vattr := vattr
+			// TODO: set the most current data from either the inband/outofband NS
+			switch vattr.Namespace {
+			case FirmwareVersionInbandNS, FirmwareVersionOutofbandNS:
+				fw := &FirmwareVersionedAttribute{}
+				if err := UnpackVersionedAttribute(&vattr, fw); err == nil {
+
```
Comment (jakeschuurmans) [inline serverservice/methods.go]: super nitpick, extra space here

**Review by jakeschuurmans — DISMISSED** [review]

File: serverservice/methods.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
+package serverservice
```
Comment (DoctorVin) [inline serverservice/methods.go:1]: Future-proof this as the `fleetdb` package?

**Review by DoctorVin — DISMISSED** [review]

File: serverservice/methods.go (hunk 2 above)
Comment (joelrebel) [reply serverservice/methods.go:1]: yep, I'll take this on separately

**Comment by joelrebel** [comment]
rebased on main

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**